### PR TITLE
Fixing ic dependendcies

### PR DESCRIPTION
**Pull request by Maximkaaa**
* Update ic_cdk to version 0.5
* Replace the dependency on `ic` with the main repo, on a fixed branch
* Remove copy-pasted types and reexport them in the ic-helpers crate

After this, other repos that need to work with ledger will not need
to specify ic dependency but can rather just use reexported types from
ic-helpers. This lets us specify the ic branch we want to use only
one time - in ic-helpers crate.

This PR is marked as draft because `ic-kit` dependency is not actually tagged yet. After https://github.com/infinity-swap/ic-kit/pull/1 is merged and tagged, this PR can also be merged (no additional edits are needed).

For @hagsteel : to make it build replace
```
ic-kit = { git = "https://github.com/infinity-swap/ic-kit", tag = "v0.4.5" }
```
with
```
ic-kit = { git = "https://github.com/infinity-swap/ic-kit", branch = "update_cdk" }
```
in `ic-helpers/Cargo.toml`

### CODE REVIEW

**Comment by Maximkaaa** [comment]
NVM, `ic-kit` was already merged. Everything is ready.

File: ic-helpers/src/factory/api.rs [hunk 1]
```diff
@@ -352,16 +349,16 @@ macro_rules! extend_with_factory_api {
...
+                    LedgerPrincipalExt, PrincipalId, Subaccount, DEFAULT_TRANSFER_FEE,
+                };
 
                 let caller = ::ic_cdk::caller();
                 let ledger = self.$state.borrow().ledger_principal();
                 let balance = ledger
                     .get_balance(
                         ::ic_cdk::api::id(),
-                        Some(Subaccount::from(&PrincipalId::from(caller))),
+                        Some(Subaccount::from(&PrincipalId(caller))),
```
Comment (hagsteel) [inline ic-helpers/src/factory/api.rs:361]:
```suggestion
                        Some(Subaccount::from(PrincipalId(caller))),
```

File: ic-helpers/src/factory/api.rs [hunk 2]
```diff
@@ -375,7 +372,7 @@ macro_rules! extend_with_factory_api {
                     &ledger,
                     caller,
                     balance,
-                    Some(Subaccount::from(&PrincipalId::from(caller))),
+                    Some(Subaccount::from(&PrincipalId(caller))),
```
Comment (hagsteel) [inline ic-helpers/src/factory/api.rs:375]:
```suggestion
                    Some(Subaccount::from(PrincipalId(caller))),
```

File: ic-helpers/src/factory/api.rs [hunk 3]
```diff
@@ -400,14 +397,14 @@ macro_rules! extend_with_factory_api {
...
-                use ::ledger_canister::account_identifier::{AccountIdentifier, Subaccount};
+                use ::ic_helpers::ledger::{AccountIdentifier, PrincipalId, Subaccount};
 
                 let factory_id = ::ic_cdk::api::id();
                 let caller = ::ic_cdk::api::caller();
-                let subaccount = Subaccount::from(&PrincipalId::from(caller));
-                let account = AccountIdentifier::from_principal(factory_id, Some(subaccount));
+                let account = AccountIdentifier::new(
+                    PrincipalId(factory_id),
+                    Some(Subaccount::from(&PrincipalId(caller))),
```
Comment (hagsteel) [inline ic-helpers/src/factory/api.rs:406]:
```suggestion
                    Some(Subaccount::from(PrincipalId(caller))),
```

**Review by hagsteel — APPROVED** [review]
Made a few suggestions for consistency but apply them as you see fit

**Comment by Maximkaaa** [comment]
@hagsteel These don't work, since Subaccount doesn't implement From<PrincipalId>.

**Comment by hagsteel** [comment]
> @hagsteel These don't work, since Subaccount doesn't implement From.

Did you try ;)

**Comment by Maximkaaa** [comment]
I sure did. These lines are inside a macro, so to see an error message, you need to try to build some factory

**Review by blutooth — APPROVED** [review]

**Pull request merged**